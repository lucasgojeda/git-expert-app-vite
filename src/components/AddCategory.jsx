import React, { useState } from 'react';
import PropTypes from 'prop-types';

/**
 * En este componente encontramos el input en el cuál se introducen las nuevas categorias a buscar.
 * @module AddCategory
 */

/**
 * @property {Function} setCategories - Sirve agregar nuevas categorias al Array de categorias a buscar, proviene del useState.
 */
const AddCategory = ({setCategories}) => {

    /**
     * El siguiente useState maneja el valor del input de busquedas.
     */
    const [inputValue, setInputValue] = useState('')

    /**
     * La siguiente función cambia el valor del inputValue cada vez que cambia el input. 
     */
    const handleInputChange = (e) => {
        setInputValue( e.target.value )
    };

    /**
     * La siguiente función guarda la nueva categoria en el Array de "categories".
     */
    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2 ){
            setCategories(cats => [ inputValue, ...cats ]);
            setInputValue('')
        };

    };

    return (
        <form onSubmit={handleSubmit}>
            <p> { inputValue } </p>
            <input
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };

export default AddCategory;