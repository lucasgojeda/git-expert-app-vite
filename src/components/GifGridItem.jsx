import React from 'react';
import PropTypes from "prop-types";

/**
 * Este componente es la "Card" donde pondremos individualmente cada gif para ser mostrado en la secciÃ³n de resultados.
 * @module GifGridItem
 */

/**
 * @property {String} title - Este es el titulo de la categoria.
 * @property {String} url - Este es el url del gif que se va a mostrar.
 */
export const GifGridItem = ( {title, url} ) => {
    return (
        <div className='card animate__animated animate__bounce'>
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    );
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};