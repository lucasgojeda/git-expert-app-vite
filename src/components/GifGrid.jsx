import React from 'react';
import "animate.css";
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

/**
 * Este componente muestra los resultados de las busquedas realizadas.
 * Desde este componente se hace el llamado a el hook que realiza la peticiÃ³n al Api.
 * @module GifGrid
 */

/**
 * 
 * 
 * @property {String} category - Nombre de la categoria a buscar.
 */
const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);


    return (
        <div className="card-grid">
            <h3>{category}</h3>
            {loading && <p>Loading</p>}
            {
                images.map(({ id, title, url }) => {
                    return <GifGridItem key={id} title={title} url={url} />
                })
            }
        </div>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};

export default GifGrid;