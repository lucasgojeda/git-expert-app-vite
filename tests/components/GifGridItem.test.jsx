import { render, screen } from "@testing-library/react";

import { GifGridItem } from "../../src/components/GifGridItem";


describe('Pruebas en el componente <GifGrifItem/>', () => {

    const title = 'saitama';
    const url = 'url';

    test('Deberia hacer match con el snapshot ', () => {

        const { container } = render(<GifGridItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifGridItem title={title} url={url} />);
        
        // screen.debug();

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( src );
        expect( alt ).toBe( title );
    });

    test('debe de mostrar el titulo en el componente', () => {

        render(<GifGridItem title={title} url={url} />);

        expect( screen.getByText( title ) ).toBeTruthy();

    });
}); 