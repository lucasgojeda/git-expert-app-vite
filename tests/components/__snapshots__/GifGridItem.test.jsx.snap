// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrifItem/> Deberia hacer match con el snapshot  1`] = `
<div>
  <div
    class="card animate__animated animate__bounce"
  >
    <img
      alt="saitama"
      src="url"
    />
    <p>
       
      saitama
       
    </p>
  </div>
</div>
`;
